juan/CP06-login

JISaldana opened the pull request:
This test verifies the correct functioning of the "Remember Me" box

JISaldana committed:
[Test] Data provider created

JISaldana committed:
[Test] Alert Message added

Alert that appears when cant login into the website added on LoginPage page

JISaldana committed:
[Test] shouldNotLoginWhenWrongPasswordIsProvided Added

Test script "shouldNotLoginWhenWrongPasswordIsProvided" was added.

This test tries to enter the website with a valid username and a wrong password.

JISaldana committed:
[Test] DataProvider added

Data provider "RememberMeLoginData" for "shouldRembemberLastLogedUsernameWhenRememberMeCheckboxIsCheck" was added

JISaldana committed:
[Test] Logout Added

Logout method for "shouldRembemberLastLogedUsernameWhenRememberMeCheckboxIsCheck" test was added on HeaderPage page

clickDeleteDataButton commented for incompatibilities and uncommented after test.

JISaldana committed:
[Test] Whitespace changes

sofiambd commented on qa-automation/src/test/java/com/digitalbank/qaautomation/tests/loginTests/shouldNotLoginWhenWrongPasswordIsProvided.java:
```diff
@@ -0,0 +1,50 @@
+        String currentUrl = driver.getCurrentUrl();
+        Assert.assertEquals(currentUrl, "http://digitalbank.upcamp.io/bank/login?error");
+
+        //Assert wrong credentials alert on login page
```
Usar steps en pages, no es necesario comentarios.

sofiambd commented on qa-automation/src/test/java/com/digitalbank/qaautomation/tests/loginTests/shouldNotLoginWhenWrongPasswordIsProvided.java:
```diff
@@ -0,0 +1,50 @@
+
+        //Assert wrong credentials alert on login page
+        String cantLoginAlert = loginPage.cantLoginAlert.getText();
+        Assert.assertEquals(cantLoginAlert, "Error Invalid credentials or access not granted due to user account status or an existing user session.\n" + "×");
```
Para strings, es buena practica definirla en el codigo y llamarla.

sofiambd commented on qa-automation/src/test/java/com/digitalbank/qaautomation/tests/loginTests/shouldRembemberLastLogedUsernameWhenRememberMeCheckboxIsCheck.java:
```diff
@@ -0,0 +1,62 @@
+
+        //Assert login
+        String currentUrl = driver.getCurrentUrl();
+        Assert.assertEquals(currentUrl, "http://digitalbank.upcamp.io/bank/home");
```
Mismo comentario que strings.

sofiambd commented on qa-automation/src/test/java/com/digitalbank/qaautomation/tests/loginTests/shouldRembemberLastLogedUsernameWhenRememberMeCheckboxIsCheck.java:
```diff
@@ -0,0 +1,62 @@
+        loginPage.logIn(user, pass);
+        HeaderPage headerPage = new HeaderPage(driver);
+
+        //Assert login
```
Evitar comentarios, con steps en las pages es suficiente.

sofiambd commented on qa-automation/src/test/java/com/digitalbank/qaautomation/pages/HeaderPage.java:
```diff
@@ -0,0 +1,39 @@
+            userProfileBtn.click();
+        }
+
+        public HomePage clickDeleteDataButton() {
```
Falta Step

sofiambd commented on qa-automation/src/test/java/com/digitalbank/qaautomation/pages/HeaderPage.java:
```diff
@@ -0,0 +1,39 @@
+            super(driver);
+        }
+
+        public void clickUserProfileBtn() {
```
Falta Step

sofiambd reviewed (changes requested):
Revisar comentarios.

JISaldana committed:
[Test] Changes

Comments was deleted, Step defined on "clickDeleteDataButton" method and
the expected URL string were defined

JISaldana committed:
Removed comments on test,
Steps added HeaderPage for "clickUserProfileBtn()" and "clickDeleteDataButton()"
Strings for expected results on tests declared.

JISaldana committed:
Merge branch 'main' of https://github.com/florencialecha/digital-bank-qa-automation into juan/CP06-login

JISaldana committed:
Public Elements changed to private, new methods defined instead

JISaldana committed:
Allure-results folder added to gitignore file

JISaldana committed:
Test changed to use methods instead of elements

JISaldana committed:
Test changed to use methods instead of elements
Assert to confirm logout successfully added

JISaldana committed:
Classes irrelevant to this test branch was deleted
(you can found it on branch "juan/CP05-login")
